src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -238,11 +246,10 @@ internal static void WriteValue(TextWriter writer, DateTimeOffset value, ODataJs
                         //
                         // ticks = *DIGIT
                         // offset = 4DIGIT
-                        string textValue = FormatDateTimeAsJsonTicksString(value);
-                        JsonValueUtils.WriteQuoted(writer, textValue);
+                        return FormatDateTimeAsJsonTicksString(value);
                     }
-
-                    break;
+                default:
+                    throw new ODataException(Strings.ODataJsonWriter_UnsupportedDateTimeFormat);

```

Comment:
Is this a breaking change?

Reply:
Or it's an internal method..

Reply:
Could you add a test when this exception gets thrown

Reply:
The thing is that code is unreachable because `ODataJsonDateTimeFormat` enum only has two members and both of those enums are accounted for by the `case` statements. In the original code, there was no `default` block. The difference is that in this version I'm returning a value and the compiler requires each code path to have a return statement, so I needed `default` block to in order to compile. And since there's nothing to return in this scenario, I just throw an exception. So if we ever add another member to that enum and not handle it, an exception will be thrown, which I believe is the right thing to do.